app/util/calldata.test.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  getAddress,
+  Interface,
+  parseEther,
+  parseUnits,
+  zeroPadValue
+} from "ethers"
+
+import { decodeExecuteParams, decodeTransferParams } from "~app/util/calldata"
+import { AccountType } from "~packages/account"
+
+describe("calldata", () => {
+  const accountAbi: string[] = [
+    "function transfer(address to, uint256 value) public returns (bool)"
+  ]
+  const accountType = AccountType.SimpleAccount
+
+  const toAddress = getAddress(zeroPadValue("0x1234", 20))
+  const tokenValue = 0.1
+  const tokenDecimals = 18
+
+  const tokenAddress = getAddress(zeroPadValue("0x5678", 20))
+  const etherValue = 0.0
+
+  const transferIface = new Interface(accountAbi)
+
+  const transferCalldata = transferIface.encodeFunctionData("transfer", [
+    toAddress,
+    parseUnits(tokenValue.toString(), tokenDecimals)
+  ])
+
+  const executeIface = new Interface([
+    "function execute(address dest, uint256 value, bytes func) external"
+  ])
+
+  const executeCalldata = executeIface.encodeFunctionData("execute", [
+    tokenAddress,
+    parseEther(etherValue.toString()),
+    transferCalldata
+  ])
```

Comment:
Try to avoid setting up all at once before tests. Let test case follow arrange, act, assert structure.

Reply:
OK! I modified it in this version using the arrange, act, assert structure:
[chg: Modify test using the arrange, act, assert structure](https://github.com/pilagod/waallet-extension/pull/156/commits/8893b9fb2012a8232cf252053c459b8109123354)